### ISSUE TITLE
`composer outdated --direct` command stopped showing version numbers and descriptions

### ISSUE THREAD

**Issue by sebastiaanluca** [composer/composer#7153]
My `composer.json`:

```json
{
    "name": "x/x",
    "description": "x",
    "keywords": [
        "laravel",
        "x"
    ],
    "license": "proprietary",
    "type": "project",
    "require": {
        "php": ">=7.2",
        "doctrine/dbal": "^2.6",
        "fideloper/proxy": "^4.0",
        "guzzlehttp/guzzle": "^6.3",
        "intervention/image": "^2.4",
        "jeffochoa/factory-stories": "^1.1",
        "jtant/laravel-env-sync": "^1.3",
        "laracasts/flash": "^3.0",
        "laravel/framework": "~5.6.7",
        "laravel/horizon": "^1.2",
        "laravel/socialite": "^3.0",
        "laravel/tinker": "^1.0",
        "laravelista/ekko": "^1.5",
        "league/csv": "^9.1",
        "league/flysystem-aws-s3-v3": "^1.0",
        "monarobase/country-list": "^2.1",
        "nesbot/carbon": "^1.22",
        "nothingworks/blade-svg": "^0.2.2",
        "nwidart/laravel-modules": "dev-master",
        "predis/predis": "^1.1",
        "propaganistas/laravel-fakeid": "^3.0",
        "roave/security-advisories": "dev-master",
        "sebastiaanluca/laravel-conditional-providers": "^1.1",
        "sebastiaanluca/laravel-helpers": "^1.0",
        "sebastiaanluca/laravel-resource-flow": "dev-develop",
        "sebastiaanluca/laravel-router": "^3.1",
        "sebastiaanluca/laravel-unbreakable-migrations": "dev-develop",
        "sentry/sentry-laravel": "dev-laravel-56-log-channels-support",
        "spatie/flysystem-dropbox": "^1.0",
        "spatie/laravel-backup": "^5.2",
        "spatie/laravel-collection-macros": "^3.7",
        "spatie/laravel-html": "^2.15",
        "spatie/laravel-permission": "^2.8",
        "spatie/once": "^2.0"
    },
    "require-dev": {
        "barryvdh/laravel-ide-helper": "^2.4",
        "filp/whoops": "^2.1",
        "fzaninotto/faker": "^1.7",
        "garygreen/pretty-routes": "^1.0",
        "itsgoingd/clockwork": "^2.0",
        "kint-php/kint": "^2.1",
        "laravel/envoy": "^1.4",
        "laravel/homestead": "^7.1",
        "mockery/mockery": "^1",
        "nunomaduro/collision": "^2.0",
        "phpunit/phpunit": "^7.0",
        "spatie/laravel-db-snapshots": "^1.1"
    },
    "repositories": {
        "laravel-modules": {
            "type": "vcs",
            "url": "https://github.com/sebastiaanluca/laravel-modules"
        },
        "sentry-laravel": {
            "type": "vcs",
            "url": "https://github.com/patrickbrouwers/sentry-laravel"
        }
    },
    "autoload": {
        "classmap": [
            "app",
            "database/seeds",
            "database/factories"
        ],
        "psr-4": {
            "App\\": "app/"
        },
        "files": [
            "app/helpers.php"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "extra": {
        "laravel": {
            "dont-discover": [
            ]
        }
    },
    "scripts": {
        "post-update-cmd": [
            "@php artisan ide-helper:generate",
            "@php artisan ide-helper:meta",
            "@php artisan ide-helper:models --nowrite"
        ],
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover"
        ]
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true,
        "platform": {
            "php": "7.2"
        }
    }
}
```

Output of `composer diagnose`:

```
Checking composer.json: WARNING
require.nwidart/laravel-modules : unbound version constraints (dev-master) should be avoided
require.roave/security-advisories : unbound version constraints (dev-master) should be avoided
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.6.3
PHP version: 7.2.0-1+ubuntu16.04.1+deb.sury.org+1
PHP binary path: /usr/bin/php7.2
```

When I run this command:

```
composer outdated --direct
```

I get the following output:

```
itsgoingd/clockwork                           v2.0.4
laravel/framework                             v5.6.5
nesbot/carbon                                 1.22.1
nwidart/laravel-modules                       dev-master 90e1557
phpunit/phpunit                               7.0.1
roave/security-advisories                     dev-master 940eb3d
spatie/laravel-backup                         5.2.2
spatie/laravel-html                           2.15.0
```

And I expected this output:

```
itsgoingd/clockwork                           v2.0.4              v2.1.1              php dev tools integrated to your browser
laravel/framework                             v5.6.5              v5.6.7              The Laravel Framework.
nesbot/carbon                                 1.22.1              1.23.0              A simple API extension for DateTime.
nwidart/laravel-modules                       dev-master 90e1557  dev-master 80c0880  Laravel Module management
phpunit/phpunit                               7.0.1               7.0.2               The PHP Unit Testing framework.
roave/security-advisories                     dev-master 940eb3d  dev-master 5eb59cd  Prevents installation of composer packag...
spatie/laravel-backup                         5.2.2               5.4.0               A Laravel 5 package to backup your appli...
spatie/laravel-html                           2.15.0              2.18.0              A fluent html builder
```

### Source of the problem

When I remove:

```
"sentry/sentry-laravel": "dev-laravel-56-log-channels-support",
```

And run the command again, it does show the newly available releases and project descriptions as expected.

**Comment by stof** [comment]
Actually, it depends on the terminal width. when the terminal is not wide enough, it drops some info to fit on a single line.

**Comment by sebastiaanluca** [comment]
Oh my. Yep, that was it 😅 `dev-laravel-56-log-channels-support` might've been too long then. A few dots to indicate it has been truncated would be nice though :p

It also seems to take the new version number and the description as one item. Name, current version, and new release fit perfectly on one line with room to spare, but because project description doesn't fit, it doesn't show the current release either. Might be worth a tweak.

**Comment by stof** [comment]
nope, it can hide any of them separately according to the code (try reducing the console width little by little.

But currently, it takes all packages into account when computing the necessary width of columns, not only the displayed ones. So this might require too much space for each column (if you have a package using `dev-laravel-56-log-channels-support` as version, the version column takes that much width even when this package is not in the displayed list)

**Comment by sebastiaanluca** [comment]
Hm, ok. But take the following example:

<img width="914" alt="screen shot 2018-03-02 at 18 45 37" src="https://user-images.githubusercontent.com/711940/36913326-db889b5e-1e49-11e8-997b-c75b4a8254b4.png">

First output is executed with a wide terminal, then resized to original width (so project description is wrapped to the next line). First 3 columns fit perfectly inside the window. Second output is run with terminal window set to original width, but it doesn't output the current release nor project description while there's enough room for the release.

Doesn't bother me though, I now know what the issue was :)

**Comment by stof** [comment]
@sebastiaanluca see my previous comment: currently, the width computation takes all packages into account, even when `--all` is not specified, and so hidden packages are increasing the column width.

**Comment by sebastiaanluca** [comment]
Ah, I see. _Hidden_ packages too. Thanks for the clarification!

**Comment by stof** [comment]
I would suggest you to reopen this issue though, as taking hidden packages into account for the computation is a bug IMO.

**Comment by sebastiaanluca** [comment]
Done. Small bug indeed :)

**Comment by Arezohayeman** [comment]
Hello-word

**Comment by Seldaek** [comment]
Fixed by #7166
